gulpfile.js

```diff
@@ -0,0 +1,221 @@
+    .pipe(dest('R/', { overwrite : true }));
+}
+
+function appendHtmlInternal() {
+  return src('R/internal.R')
+    .pipe(print())
+    .pipe(footer("# dashHtmlComponents metadata appended by `npm unify` command\n"))
+    .pipe(footer(fs.readFileSync('gulp-assets/dash-html-components/R/internal.R')))
+    .pipe(dest('R/', { overwrite: true }));
+}
+
+function appendTableInternal() {
+  return src('R/internal.R')
+    .pipe(print())
+    .pipe(footer("# dashTable metadata appended by `npm unify` command\n"))
+    .pipe(footer(fs.readFileSync('gulp-assets/dash-table/R/internal.R')))
+    .pipe(dest('R/', { overwrite: true }));
+}
+
+// Remove component package imports from NAMESPACE.
+function replaceComponentImports() {
+  return src('./NAMESPACE')
+    .pipe(print())
+    .pipe(replace(/import\(dashCoreComponents\)/, '#import(dashCoreComponents)'))
+    .pipe(replace(/import\(dashHtmlComponents\)/, '#import(dashHtmlComponents)'))
+    .pipe(dest('.'));
+}
+
+// Remove component package imports and collate from DESCRIPTION.
+function replaceDescriptionImports() {
+  return src('./DESCRIPTION')
+    .pipe(print())
+    .pipe(replace(/  dashHtmlComponents \(== 1.1.1\),\n/, ''))
+    .pipe(replace(/  dashCoreComponents \(== 1.13.0\),\n/, ''))
+    .pipe(replace(/  dashTable \(== 4.11.0\),\n/, ''))
+    .pipe(replace(/Collate:/, '#Collate:'))
+    .pipe(dest('.'));
+}
+
+// Point dependency sourcing to Dash package.
+function replacePackageDependency() {
+  return src('R/internal.R')
+    .pipe(print())
+    .pipe(replace(/package = "dashCoreComponents"/g, 'package = "dash"'))
+    .pipe(replace(/package = "dashHtmlComponents"/g, 'package = "dash"'))
+    .pipe(replace(/package = "dashTable"/g, 'package = "dash"'))
+    .pipe(dest('R/'));
+}
+
+// Clean unification changes from NAMESPACE.
+function cleanNamespace() {
+  return src('./NAMESPACE')
+    .pipe(print())
+    .pipe(replace(/#import\(dashCoreComponents\)/, 'import(dashCoreComponents)'))
+    .pipe(replace(/#import\(dashHtmlComponents\)/, 'import(dashHtmlComponents)'))
+    .pipe(replace(/# dashCoreComponents exports.*/s, ''))
+    .pipe(dest('.'));
+}
+
+// Clean unification changes from DESCRIPTION.
+function cleanDescriptionImports() {
+  return src('./DESCRIPTION')
+    .pipe(print())
+    .pipe(replace(/Imports:/, 'Imports:\n  dashHtmlComponents (== 1.1.1),\n  dashCoreComponents (== 1.13.0),\n  dashTable (== 4.11.0),'))
+    .pipe(replace(/#Collate:/, 'Collate:'))
+    .pipe(dest('.'));
+}
+
+// Clean unification changes from internal.R.
+function cleanInternal() {
+  return src('R/internal.R')
+    .pipe(print())
+    .pipe(replace(/# dashCoreComponents metadata.*/s, ''))
+    .pipe(dest('R/'));
+}
+
+// Clean component function files from R directory.
+function cleanR() {
+  return del([
+    'R/(dcc|html|dashDataTable)*.R',
+    'man/(dcc|html|dashCoreComponents|dashHtmlComponents|dashDataTable|dashTable)*.Rd'
+  ]);
+}
```

Comment:
I wonder if we could modify this step if we copied the contents of `dashTable`, `dashHtmlComponents`, `dashCoreComponents` directly into the correct locations within the cloned `plotly/dashR` repo, then (optionally) delete the three original packages after the merge is complete. This looks like we're cleaning up artifacts individually rather than just :articulated_lorry: them over.

Reply:
Done in 2ee8b56